# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaManager.cpp
# Commit: f8b5be61f5f8
# Full Hash: f8b5be61f5f8a0c9c53957ecf23d81be1c2130bd
# Author: Paul Adenot <paul@paul.cx>
# Date: 2020-07-08 21:49:35
# Description:
#   Bug 1650181 - Null-check when getting default output device info, because there can be no output device. r=achronop
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D82560
# ==============================================================================

diff -r 2df443d3fcd0 -r f8b5be61f5f8 dom/media/MediaManager.cpp
--- a/dom/media/MediaManager.cpp	Wed Jul 08 09:04:03 2020 +0000
+++ b/dom/media/MediaManager.cpp	Wed Jul 08 09:07:48 2020 +0000
@@ -4436,7 +4436,8 @@
                     RefPtr<AudioDeviceInfo> outputDevice =
                         enumerator->DefaultDevice(
                             CubebDeviceEnumerator::Side::OUTPUT);
-                    if (outputDevice->GroupID().Equals(inputDeviceGroupId)) {
+                    if (outputDevice &&
+                        outputDevice->GroupID().Equals(inputDeviceGroupId)) {
                       LOG("Device group id match when %s, "
                           "not turning the input device off (%s)",
                           aEnable ? "unmuting" : "muting",
